wire/error.go

```diff
@@ -9,6 +9,176 @@ import (
 	"fmt"
 )
 
+// ErrorCode describes a kind of message error.
+type ErrorCode int
+
+// These constants are used to identify a specific Error.
+const (
+	// ErrNonCanonicalVarInt is returned when a variable length integer
+	// is encoded non-canonically.
```

Comment:
I would probably reverse the wording here because there isn't a single "non-canonical encoding" as this implies.  There is, however, only one canonical encoding.

e.g.
`... when a variable length integer is not canonically encoded.`